### ISSUE TITLE
add Markdown rendering component

### ISSUE THREAD

**Issue by Groostav** [TomasMikula/RichTextFX#348]
> As a software developer using javaFX, I want an entirely textual way to represent large text documents, so that I may easily create things like in-application "help", "about", and other text-heavy content.

So, the `Text`, `TextFlow`, `TextField`, and etc objects really aren't tailored to creating long running text content. If you spend a great deal of time with these objects, using custom CSS and associated style classes I'd be willing to bet you could come up with something that looks as good as this issue does on github. But I wrote this issue on github with all its associated block-quotes, mono-space things, etc in less than 10 minutes, and generating that same component using the aforementioned javafx components would take an order of magnitude more time.

So I want Markdown in javafx, I am aware of [MarkdownWriterFX](https://github.com/JFormDesigner/markdown-writer-fx), but I don't want to have to boot a web rendering component (includes nashorn?) to do it, and I want things like `[mailto: links](mailto: issue1234@github.com)` to work (--which cant be done using the above markdown component without annoying hacks)

Now, given some of the (ongoing) work on [AnTextFX](../../../EmpowerOperations/AnTextFX), I think all of the in-paragraph highlighting features can be done very easily with ANTLR and RichTextFX. The multi paragraph features, such as indentation changes, list-views, etc may be more difficult. This also doesn't include support for tables, which I see as a nice to have but not a necessity for this component to deliver some level of value.

Specifically, I would like to deploy an application with something like the following:

``` fxml
<!-- content view.fxml -->

<VBox fx:controller="com.mycompany.ContentController">
  <children>
    <RichTextFXMarkdown content="/com/mycompany/content.md"/>
  </children>
</VBox>

```

``` md
#content.md
--

#About OurProject

OurProject is the _best_ project on the face of the earth. It solves all your problems! Everything is better when you purchase OurProject, including your ability to do things like write issues on [github](github.com).

Please contact [support@mycompany.com](mailto:support@mycompany.com) with any questions. Remember, most bugs are actually features, so only send us email if you're certain you've actually got a bug!

And remember, [looking for bugs in our software constitutes a violation of our license agreement and is tantamount to a federal crime!](https://www.scribd.com/doc/274176394/No-You-Really-Can-t-Mary-Ann-Davidson-Blog)


```

Would anybody else be interested in such a component?
## 

**tl;dr** Implement custom markdown rendering component using a parser and RichTextFX. New repo? Does anybody else care to join me?

**Comment by JordanMartinez** [comment]
Closing due to being outside the scope of this project. I've also written an explanation of what "out of the scope of this project" means in [the wiki](https://github.com/TomasMikula/RichTextFX/wiki/General-Guidelines#why-dont-we-add-feature-x-to-richtextfx)
